### PR TITLE
Make DataFrameComparer generic a generic Dataset

### PR DESCRIPTION
**Pull request by MrPowers**

### CODE REVIEW

File: src/main/scala/com/github/mrpowers/spark/fast/tests/DatasetComparer.scala [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        throw new DatasetContentMismatch(countMismatchMessage(actualCount, expectedCount))
+      }
+
+      val expectedIndexValue: RDD[(Long, U)] = zipWithIndex(actualDS.rdd)
+      val resultIndexValue: RDD[(Long, U)] = zipWithIndex(expectedDS.rdd)
+      val unequalRDD = expectedIndexValue
+        .join(resultIndexValue)
+        .filter {
+          case (idx, (o1, o2)) =>
+            !o1.equals(o2)
```
Comment (snithish) [inline src/main/scala/com/github/mrpowers/spark/fast/tests/DatasetComparer.scala]: @MrPowers Absence of RowComparer.areRowsEqual(r1, r2, 0.0)), would mean relying on equals method on Row class, are we okay with that?

**Comment by snithish** [comment]
@MrPowers I have raised #7 with a few clean ups, nice work on using Generics for Dataset equality.

File: src/main/scala/com/github/mrpowers/spark/fast/tests/DatasetComparer.scala (hunk 1 above)
Comment (MrPowers) [inline src/main/scala/com/github/mrpowers/spark/fast/tests/DatasetComparer.scala]: I think so.... haha.  I am going to merge this pull request in an open an issue to discuss this issue in more detail!

Comment (snithish) [inline src/main/scala/com/github/mrpowers/spark/fast/tests/DatasetComparer.scala]: @MrPowers That would be great, I am thinking of having a `assertDatasetLargeEquality` which would also take in a comparer function, meaning users of API can define their own equality logic.

**Pull request merged**